# **Introduction**


Due to the fast paced changes across industries due to technology advancements ,most companies are working on making sure that they become the best in every competitive business fields. As part of this strategy ,our company is expanding in to new industries to diversify its portfolio including the aviation industry by providing both commercial and private aircraft services. Venturing into this type of industry might be a bit challenging due to some of the risk factors invovled and limited  knowledge on the best aircrafts models  to purchase based on their safety level and lower accident risks.

This project focuses to analyze aviation accident data to provide clear and accurate findings for decision making in regards to the purchase of the aircrafts.Using data analysis tecniques,we will be able to identify the trends and patterns of the accident risks and use our findings to help decide which aircraft to purchase.

# **1.Business Understanding**

Stakeholders

1.Head of Aviation Division:The primary decision maker to determine aircrafts to purchase

2.Shareholders:Assess risk before investment, inform strategic decisions, protect financial interests, and monitor safety performance.

3.Rgulatory body:Use findings to identify high-risk aircraft, inform safety policies, and focus oversight on areas with higher risk.


4.Suppliers:Identify high-risk models, improve product design, support safety communication, and align with operators needs.

5.Operations Team:Use our findings to enforcing regulations, safety oversight and training.








# Business Value

This analysis provide a number of benefits to our company which include:

1.Future Accident prevention by understanding factors contributing to the accidents and their root causes.

2.Cost savings which include preventing loss of life,aircraft damages,and operations disruptions.

3.Regulatory compliance by adhering to standards set by bodies and ensuring legal compliance and helps organizations meet established safety requirements.

4.Continuous Improvement through Risk analysis which mandates continuous monitoring and review of safety performance to adapt to new challenges and emerging risks.

5.Data-Driven Decision Making as the process yields evidence-based insights that enable aviation professionals and regulators to make informed decisions regarding safety improvements and resource allocation.

 # **Data Understanding**

In this analysis we will use data from the National Transportation Safety Board that includes aviation accident data from 1962 to 2023 about civil aviation accidents and selected incidents in the United States and international waters.From this analysis we will use statistical analysis to summarize, explore, and visualize patterns in the data.

# Data Exploration
In this section we will get to have a summary of our dataset,the number of rows and columns,a glimpse of the entries to understand what we are working with and what we need to make our analysis.

In [201]:
#importing python libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [202]:
#loading the dataset
df = pd.read_csv('../Data/AviationData.csv',encoding='latin1',low_memory=False) 

# Print number of rows and columns
print(f"The number of rows is {df.shape[0]} and the number of columns is {df.shape[1]}.\n")



The number of rows is 88889 and the number of columns is 31.



In [203]:
# View the first five rows
print("Displaying the first five rows of the dataset:")
df.head()

Displaying the first five rows of the dataset:


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [204]:
# View the data structure
print("Displaying the data structure and info:")
df.info()

Displaying the data structure and info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14

In [205]:
print("Displaying summary statistics of numerical columns:")
df.describe()

Displaying summary statistics of numerical columns:


Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

From the above we have a summary of dataset :

1. The dataset has 88,889 rows and 31 columns.
2. We have 5 numerical columns and 26 categorical columns.
3. Most accidents report zero fatalities and injuries based on the median values of 0 across injury columns.
4. The majority of aircraft involved in accidents are single-engine aircraft, as indicated by a median of 1 engine.



In [206]:
# Basic analysis of categorical columns

# Weather.Condition
print("Weather.Condition value counts:")
print(df['Weather.Condition'].value_counts(dropna=False))

# Injury.Severity
print("Injury.Severity value counts:")
print(df['Injury.Severity'].value_counts(dropna=False))



Weather.Condition value counts:
Weather.Condition
VMC    77303
IMC     5976
NaN     4492
UNK      856
Unk      262
Name: count, dtype: int64
Injury.Severity value counts:
Injury.Severity
Non-Fatal     67357
Fatal(1)       6167
Fatal          5262
Fatal(2)       3711
Incident       2219
              ...  
Fatal(80)         1
Fatal(217)        1
Fatal(169)        1
Fatal(88)         1
Fatal(189)        1
Name: count, Length: 110, dtype: int64


# Data Cleaning
In this part we will check fo missing values,duplicates and outliers and select the columns that are relevant to our analysis.

In [207]:
#check missing values in dataset
missing_values=df.isna().sum().sort_values(ascending=False)
missing_values

Schedule                  76307
Air.carrier               72241
FAR.Description           56866
Aircraft.Category         56602
Longitude                 54516
Latitude                  54507
Airport.Code              38757
Airport.Name              36185
Broad.phase.of.flight     27165
Publication.Date          13771
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Fatal.Injuries      11401
Engine.Type                7096
Report.Status              6384
Purpose.of.flight          6192
Number.of.Engines          6084
Total.Uninjured            5912
Weather.Condition          4492
Aircraft.damage            3194
Registration.Number        1382
Injury.Severity            1000
Country                     226
Amateur.Built               102
Model                        92
Make                         63
Location                     52
Investigation.Type            0
Event.Date                    0
Accident.Number               0
Event.Id                      0
dtype: i

## Cleaning Weather Conditions
The Weather.Condition variable has labels such as 'UNK','Unk' and Nan(missing values) 
These categories are standardized into label'Unknown'

In [208]:
#Standardize categorical values
df['Weather.Condition'] = df['Weather.Condition'].replace({
    'UNK': 'Unknown',
    'Unk': 'Unknown',
    'Nan':'Unknown'
})
df['Weather.Condition'].value_counts()



Weather.Condition
VMC        77303
IMC         5976
Unknown     1118
Name: count, dtype: int64

# Cleaning Injury Severity
In this section we will categorize the injury severity into two which are 'fatal and 'non_fatal' to simplify the column for more clarity and easy understsnding on the level of severity of injuries.

In [209]:

# Step 1: Standardize capitalization and strip spaces
df['Injury.Severity'] = df['Injury.Severity'].astype(str).str.strip().str.capitalize()

# Step 2: Create Grouped_injury column
def clean_injury(x):
    if 'Fatal' in x:          # catches Fatal, Fatal(1), Fatal(2), etc.
        return 'Fatal'
    elif x.lower() == 'nan':   # handle NaN values converted to string
        return 'Unknown'
    else:
        return 'Non-Fatal'

df['Grouped_injury'] = df['Injury.Severity'].apply(clean_injury)

# Step 3: Check counts
print("Grouped_injury counts:")
print(df['Grouped_injury'].value_counts())


Grouped_injury counts:
Grouped_injury
Non-Fatal    70063
Fatal        17826
Unknown       1000
Name: count, dtype: int64


From the above step we can conclude that the reported number of Non fatal injuries(70063) are more as compared to reported fatal injuries(17826) in the aircraft accidents.

# Clean Aircraft specific data 

In [ ]:
#select specific columns to work with
cols = [
    'Make',
    'Model',
    'Engine.Type',
    'Purpose.of.flight',
    'Aircraft.damage',
    'Grouped_injury'
]

aircraft_df = df[cols].copy()
aircraft_df


Make      Model    Engine.Type Purpose.of.flight  \
0                         Stinson      108-3  Reciprocating          Personal   
1                           Piper   PA24-180  Reciprocating          Personal   
2                          Cessna       172M  Reciprocating          Personal   
3                        Rockwell        112  Reciprocating          Personal   
4                          Cessna        501            NaN          Personal   
...                           ...        ...            ...               ...   
88884                       PIPER  PA-28-151            NaN          Personal   
88885                    BELLANCA       7ECA            NaN               NaN   
88886  AMERICAN CHAMPION AIRCRAFT      8GCBC            NaN          Personal   
88887                      CESSNA       210N            NaN          Personal   
88888                       PIPER  PA-24-260            NaN          Personal   

      Aircraft.damage Grouped_injury  
0           Destroyed          Fatal  
1           Destroyed          Fatal  
2           Destroyed          Fatal  
3           Destroyed          Fatal  
4           Destroyed          Fatal  
...               ...            ...  
88884             NaN      Non-Fatal  
88885             NaN        Unknown  
88886     Substantial      Non-Fatal  
88887             NaN        Unknown  
88888             NaN      Non-Fatal  

[88889 rows x 6 columns]

In [ ]:
#clean text columns to avoid duplicates
text_cols = ['Make', 'Model', 'Engine.Type', 'Purpose.of.flight', 'Aircraft.damage']

for col in text_cols:
    aircraft_df[col] = (
        aircraft_df[col]
        .astype(str)
        .str.strip()
        .str.upper()
        .replace('NAN', 'UNKNOWN')
    )


In [214]:
#group engine types
def engine_group(engine):
    if 'RECIPROCATING' in engine:
        return 'RECIPROCATING'
    elif 'TURBINE' in engine:
        return 'TURBINE'
    elif 'TURBOFAN' in engine:
        return 'TURBOFAN'
    elif 'TURBOPROP' in engine:
        return 'TURBOPROP'
    else:
        return 'OTHER / UNKNOWN'

aircraft_df['Engine_Group'] = aircraft_df['Engine.Type'].apply(engine_group)


In [218]:
#Group aicraft damage according to severity
def damage_group(damage):
    if damage in ['DESTROYED']:
        return 'HIGH DAMAGE'
    elif damage in['SUBSTANTIAL']:
        return 'AVERAGE DAMAGE'
    elif damage in ['MINOR']:
        return 'LOW DAMAGE'
    else:
        return 'UNKNOWN'

aircraft_df['Damage_Group'] = aircraft_df['Aircraft.damage'].apply(damage_group)
